src/sst/core/config.cc

```diff
@@ -41,163 +43,826 @@ using namespace std;
+        &ConfigHelper::setWritePartition, &ConfigHelper::setWritePartitionFile, true),
+
+    /* Advanced Features */
+    DEF_SECTION_HEADING("Advanced Options"),
+#ifdef SST_CONFIG_HAVE_MPI
+    DEF_FLAG(
+        "parallel-load", 0, "Enable parallel loading of configuration. This option is ignored for single rank jobs.",
+        &ConfigHelper::enableParallelLoad),
+#endif
+    DEF_ARG(
+        "timeVortex", 0, "MODULE", "Select TimeVortex implementation <lib.timevortex>", &ConfigHelper::setTimeVortex,
+        true),
+    DEF_FLAG_OPTVAL(
+        "interthread-links", 0, "[EXPERIMENTAL] Set whether or not interthread links should be used <false>",
+        &ConfigHelper::setInterThreadLinks, &ConfigHelper::setInterThreadLinksArg, true),
+    DEF_ARG("debug-file", 0, "FILE", "File where debug output will go", &ConfigHelper::setDebugFile, true),
+    DEF_ARG("lib-path", 0, "LIBPATH", "Component library path (overwrites default)", &ConfigHelper::setLibPath, true),
+    DEF_ARG(
+        "add-lib-path", 0, "LIBPATH", "Component library path (appends to main path)", &ConfigHelper::addLibPath, true),
+    DEF_ARG("run-mode", 0, "MODE", "Set run mode [ init | run | both (default)]", &ConfigHelper::setRunMode, true),
 #ifdef USE_MEMPOOL
-    DEF_ARGOPT(
-        "output-undeleted-events", "FILE",
+    DEF_ARG(
+        "output-undeleted-events", 0, "FILE",
         "file to write information about all undeleted events at the end of simulation (STDOUT and STDERR can be used "
         "to output to console)",
-        &Config::setWriteUndeleted),
+        &ConfigHelper::setWriteUndeleted, true),
 #endif
-    DEF_ARGOPT(
-        "model-options", "STR",
-        "provide options to the python configuration script.  Additionally, any arguments provided after -- will be "
-        "used as model options (Note that using -- will override anything supplied in --model-options).",
-        &Config::setModelOptions),
-    DEF_ARGOPT_SHORT("num_threads", 'n', "NUM", "number of parallel threads to use per rank", &Config::setNumThreads),
-    { { nullptr, 0, nullptr, 0 }, nullptr, nullptr, nullptr, nullptr }
+
+    /* Advanced Features - Environment Setup/Reporting */
+    DEF_SECTION_HEADING("Advanced Options - Environment Setup/Reporting"),
+    DEF_FLAG("print-env", 0, "Print environment variables SST will see", &ConfigHelper::enablePrintEnv),
+    DEF_FLAG("disable-signal-handlers", 0, "Disable signal handlers", &ConfigHelper::disableSigHandlers),
+    DEF_FLAG("no-env-config", 0, "Disable SST environment configuration", &ConfigHelper::disableEnvConfig),
+    { { nullptr, 0, nullptr, 0 }, nullptr, nullptr, nullptr, nullptr, false, false }
 };
 static const size_t nLongOpts = (sizeof(sstOptions) / sizeof(sstLongOpts_s)) - 1;
 
+
 bool
-Config::usage()
+ConfigHelper::parseBoolean(const std::string& arg, bool& success, const std::string& option)
+{
+    success = true;
+
+    std::string arg_lower(arg);
```

Comment:
If the very first thing we do is copy the string we might as well just capture it by value and operate on the value.

Reply:
I need to keep the original around for the error message.